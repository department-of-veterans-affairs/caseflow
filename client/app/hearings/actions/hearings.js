import * as Constants from '../constants/constants';

export const handleServerError = (err) => ({
  type: Constants.HANDLE_SERVER_ERROR,
  payload: {
    err
  }
});
