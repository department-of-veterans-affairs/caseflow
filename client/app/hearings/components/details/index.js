export * from './Address';
export * from './ReadOnly';
