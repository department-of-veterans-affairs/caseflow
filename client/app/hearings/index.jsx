import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import perflogger from 'redux-perf-middleware';
import thunk from 'redux-thunk';

import ConfigUtil from '../util/ConfigUtil';
import DocketsContainer from './DocketsContainer';
import DailyDocketContainer from './DailyDocketContainer';
import HearingWorksheetContainer from './HearingWorksheetContainer';
import { hearingsReducers, mapDataToInitialState } from './reducers/index';
import ScrollToTop from './util/ScrollTop';

const configureStore = (data) => {

  const middleware = [];

  if (!ConfigUtil.test()) {
    middleware.push(thunk, perflogger);
  }

  // This is to be used with the Redux Devtools Chrome extension
  // https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd
  // eslint-disable-next-line no-underscore-dangle
  const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

  const initialData = mapDataToInitialState(data);
  const store = createStore(
    hearingsReducers,
    initialData,
    composeEnhancers(applyMiddleware(...middleware))
  );

  if (module.hot) {
    // Enable Webpack hot module replacement for reducers
    module.hot.accept('./reducers/index', () => {
      store.replaceReducer(hearingsReducers);
    });
  }

  return store;
};

const Hearings = ({ hearings }) => {

  return <Provider store={configureStore(hearings)}>
    <div>
      <BrowserRouter>
        <div>
         <ScrollToTop />
          <Route exact path="/hearings/dockets"
            component={() => (
              <DocketsContainer
                veteran_law_judge={hearings.veteran_law_judge} />
            )}
          />

          <Route exact path="/hearings/dockets/:date"
            component={(props) => (
              <DailyDocketContainer
                veteran_law_judge={hearings.veteran_law_judge}
                date={props.match.params.date} />
            )}
          />

        <Route exact path="/hearings/:hearingId/worksheet"
            component={(props) => (
              <HearingWorksheetContainer
                veteran_law_judge={hearings.veteran_law_judge}
                hearingId={props.match.params.hearingId} />
            )}
          />
        </div>

      </BrowserRouter>
    </div>
  </Provider>;
};

export default Hearings;
