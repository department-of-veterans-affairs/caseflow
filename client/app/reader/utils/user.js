// External Dependencies
import { some } from 'lodash';

/**
 * Helper Method to Determine if the User is editing text
 */
export const isUserEditingText = () => some(
  document.querySelectorAll('input,textarea'),
  (elem) => document.activeElement === elem
);
