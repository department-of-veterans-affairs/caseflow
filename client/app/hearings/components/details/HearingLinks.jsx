import { css } from 'glamor';
import PropTypes from 'prop-types';
import React from 'react';

import { COLORS } from '../../../constants/AppConstants';
import { VIRTUAL_HEARING_HOST, virtualHearingRoleForUser } from '../../utils';
import {
  rowThirds,
  labelPadding,
  labelPaddingFirst,
  hearingLinksContainer
} from './style';
import COPY from '../../../../COPY';
import CopyTextButton from '../../../components/CopyTextButton';
import VirtualHearingLink from '../VirtualHearingLink';

export const VirtualHearingLinkDetails = ({
  aliasWithHost,
  pin,
  role,
  link,
  hearing,
  wasVirtual,
  isVirtual,
  user,
  label,
  virtualHearing
}) => (
  <React.Fragment>
    {hearing?.scheduledForIsPast || wasVirtual ? (
      <span>N/A</span>
    ) : (
      <VirtualHearingLink
        label={label}
        user={user}
        virtualHearing={virtualHearing}
        isVirtual={isVirtual}
        link={link}
        hearing={hearing}
      />
    )}
    {hearing.conferenceProvider === 'pexip' ?
      (<><div {...labelPaddingFirst}>
        <strong>Conference Room: </strong>
        {`${aliasWithHost}`}
      </div>
      <div {...labelPadding}>
        <strong>PIN: </strong>
        {pin}
      </div></>) : (<div {...labelPaddingFirst} className="helper-text">
        {link}
      </div>)
    }
    {!hearing?.scheduledForIsPast && !wasVirtual && (
      <CopyTextButton ariaLabel={`Copy ${role} Link`} text={`Copy ${role} Link`} textToCopy={link} />
    )}
  </React.Fragment>
);

VirtualHearingLinkDetails.propTypes = {
  aliasWithHost: PropTypes.string,
  pin: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string
  ]),
  role: PropTypes.string,
  link: PropTypes.string,
  hearing: PropTypes.object,
  wasVirtual: PropTypes.bool,
  isVirtual: PropTypes.bool,
  user: PropTypes.object,
  label: PropTypes.string,
  virtualHearing: PropTypes.object
};

export const LinkContainer = (
  { link, linkText, user, hearing, isVirtual, wasVirtual, virtualHearing, role, label }
) => (
  <div id={`${role.toLowerCase()}-hearings-link`} {...css({ marginTop: '1.5rem' })}>
    <strong>{label}: </strong>
    {!virtualHearing || virtualHearing?.status === 'pending' ? (
      <span {...css({ color: COLORS.GREY_MEDIUM })}>{COPY.VIRTUAL_HEARING_SCHEDULING_IN_PROGRESS}</span>
    ) : (
      <VirtualHearingLinkDetails
        label={linkText}
        user={user}
        virtualHearing={virtualHearing}
        isVirtual={isVirtual}
        wasVirtual={wasVirtual}
        hearing={hearing}
        link={link}
        role={role}
        aliasWithHost={virtualHearing?.aliasWithHost}
        pin={role === 'VLJ' ? virtualHearing?.hostPin : virtualHearing?.guestPin}
      />
    )}
  </div>
);

LinkContainer.propTypes = {
  hearing: PropTypes.object,
  isVirtual: PropTypes.bool,
  label: PropTypes.string,
  link: PropTypes.string,
  linkText: PropTypes.string,
  role: PropTypes.string,
  user: PropTypes.object,
  virtualHearing: PropTypes.object,
  wasVirtual: PropTypes.bool
};

export const HearingLinks = ({ hearing, virtualHearing, isVirtual, wasVirtual, user }) => {
  if (!isVirtual && !wasVirtual) {
    return null;
  }

  const showHostLink = virtualHearingRoleForUser(user, hearing) === VIRTUAL_HEARING_HOST;

  return (
    <div {...rowThirds} {...hearingLinksContainer}>
      {showHostLink && <LinkContainer
        hearing={hearing}
        isVirtual={isVirtual}
        label={COPY.VLJ_VIRTUAL_HEARING_LINK_LABEL}
        link={virtualHearing?.hostLink}
        linkText={COPY.VLJ_VIRTUAL_HEARINGS_LINK_TEXT}
        role="VLJ"
        user={user}
        virtualHearing={virtualHearing}
        wasVirtual={wasVirtual}
      />}
      <LinkContainer
        hearing={hearing}
        isVirtual={isVirtual}
        label={COPY.GUEST_VIRTUAL_HEARING_LINK_LABEL}
        link={virtualHearing?.guestLink}
        linkText={COPY.GUEST_VIRTUAL_HEARINGS_LINK_TEXT}
        role="Guest"
        user={user}
        virtualHearing={virtualHearing}
        wasVirtual={wasVirtual}
      />
      <div />
    </div>
  );
};

HearingLinks.propTypes = {
  user: PropTypes.object,
  hearing: PropTypes.object,
  isVirtual: PropTypes.bool,
  wasVirtual: PropTypes.bool,
  virtualHearing: PropTypes.object
};
