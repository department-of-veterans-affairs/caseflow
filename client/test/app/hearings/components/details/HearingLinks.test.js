import React from 'react';

import { HearingLinks } from 'app/hearings/components/details/HearingLinks';
import { anyUser, vsoUser } from 'test/data/user';
import { inProgressvirtualHearing } from 'test/data/virtualHearings';
import { virtualHearing, amaHearing } from 'test/data/hearings';
import { mount } from 'enzyme';
import VirtualHearingLink from
  'app/hearings/components/VirtualHearingLink';

const hearing = {
  scheduledForIsPast: false
};

describe('HearingLinks', () => {
  test('Matches snapshot with default props when passed in', () => {
    const form = mount(
      <HearingLinks />
    );

    expect(form).toMatchSnapshot();
    expect(form.find(VirtualHearingLink)).toHaveLength(0);
  });

  test('Matches snapshot when hearing is virtual and in progress', () => {
    const form = mount(
      <HearingLinks
        hearing={hearing}
        isVirtual
        user={anyUser}
        virtualHearing={inProgressvirtualHearing}
      />
    );

    expect(form).toMatchSnapshot();
    expect(form.find(VirtualHearingLink)).toHaveLength(2);
    expect(
      form.find(VirtualHearingLink).exists({ label: 'Join Virtual Hearing' })
    ).toBe(true);
    expect(
      form.find(VirtualHearingLink).exists({ label: 'Start Virtual Hearing' })
    ).toBe(true);
  });

  test('Matches snapshot when hearing was virtual and occurred', () => {
    const form = mount(
      <HearingLinks
        hearing={hearing}
        wasVirtual
        user={anyUser}
        virtualHearing={inProgressvirtualHearing}
      />
    );

    expect(form).toMatchSnapshot();
    expect(form.find(VirtualHearingLink)).toHaveLength(0);
    expect(
      form.find('span').filterWhere((node) => node.text() === 'Expired')
    ).toHaveLength(2);
  });

  test('Only displays Guest Link when user is not a host', () => {
    const form = mount(
      <HearingLinks
        hearing={amaHearing}
        isVirtual
        user={vsoUser}
        virtualHearing={virtualHearing.virtualHearing}
      />
    );

    expect(form).toMatchSnapshot();
    expect(form.find(VirtualHearingLink)).toHaveLength(1);
    // Ensure it's the guest link
    expect(form.find(VirtualHearingLink).prop('link')).toEqual(amaHearing.virtualHearing.guestLink)
  })
});
