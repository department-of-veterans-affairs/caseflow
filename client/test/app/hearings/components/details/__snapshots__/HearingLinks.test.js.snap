// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HearingLinks Matches snapshot when hearing is virtual and in progress 1`] = `
<HearingLinks
  hearing={
    Object {
      "scheduledForIsPast": false,
    }
  }
  isVirtual={true}
  user={
    Object {
      "addressCity": "Washington",
      "addressLine1": "123 Fake St.",
      "addressState": "DC",
      "addressZip": "20001",
      "name": "John Smith",
    }
  }
  virtualHearing={
    Object {
      "aliasWithHost": "BVA#00000001@localhost",
      "clientHost": "localhost",
      "guestPin": "12345678#",
      "hostPin": "1234567890#",
      "jobCompleted": true,
      "status": "active",
    }
  }
>
  <div
    data-css-19plkw7=""
    data-css-1fmltfu=""
  >
    <LinkContainer
      hearing={
        Object {
          "scheduledForIsPast": false,
        }
      }
      isVirtual={true}
      label="VLJ Link"
      linkText="Start Virtual Hearing"
      role="VLJ"
      user={
        Object {
          "addressCity": "Washington",
          "addressLine1": "123 Fake St.",
          "addressState": "DC",
          "addressZip": "20001",
          "name": "John Smith",
        }
      }
      virtualHearing={
        Object {
          "aliasWithHost": "BVA#00000001@localhost",
          "clientHost": "localhost",
          "guestPin": "12345678#",
          "hostPin": "1234567890#",
          "jobCompleted": true,
          "status": "active",
        }
      }
    >
      <div
        data-css-tx10g=""
        id="vlj-hearings-link"
      >
        <strong>
          VLJ Link
          : 
        </strong>
        <VirtualHearingLinkDetails
          aliasWithHost="BVA#00000001@localhost"
          hearing={
            Object {
              "scheduledForIsPast": false,
            }
          }
          isVirtual={true}
          label="Start Virtual Hearing"
          pin="1234567890#"
          role="VLJ"
          user={
            Object {
              "addressCity": "Washington",
              "addressLine1": "123 Fake St.",
              "addressState": "DC",
              "addressZip": "20001",
              "name": "John Smith",
            }
          }
          virtualHearing={
            Object {
              "aliasWithHost": "BVA#00000001@localhost",
              "clientHost": "localhost",
              "guestPin": "12345678#",
              "hostPin": "1234567890#",
              "jobCompleted": true,
              "status": "active",
            }
          }
        >
          <VirtualHearingLink
            hearing={
              Object {
                "scheduledForIsPast": false,
              }
            }
            isVirtual={true}
            label="Start Virtual Hearing"
            newWindow={true}
            showFullLink={false}
            user={
              Object {
                "addressCity": "Washington",
                "addressLine1": "123 Fake St.",
                "addressState": "DC",
                "addressZip": "20001",
                "name": "John Smith",
              }
            }
            virtualHearing={
              Object {
                "aliasWithHost": "BVA#00000001@localhost",
                "clientHost": "localhost",
                "guestPin": "12345678#",
                "hostPin": "1234567890#",
                "jobCompleted": true,
                "status": "active",
              }
            }
          >
            <Link
              disabled={false}
              target="_blank"
            >
              <a
                target="_blank"
                type={null}
              >
                <strong>
                  Start Virtual Hearing
                </strong>
                <span
                  data-css-32srrr=""
                >
                   
                  <ExternalLink
                    fill="#0071bc"
                  >
                    <svg
                      height="16px"
                      version="1.1"
                      viewBox="0 0 18 16"
                      width="18px"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        fill="none"
                        fillRule="evenodd"
                        stroke="none"
                        strokeWidth="1"
                      >
                        <g
                          fill="#0071bc"
                          fillRule="nonzero"
                        >
                          <g
                            transform="translate(1.000000, 1.000000)"
                          >
                            <path
                              d="M12.2767813,8.17242825 L11.7057188,8.17242825 C11.6223125,8.17242825 11.553875,8.19969476 11.500625,8.25448292 C11.4469375,8.30895216 11.4202188,8.37876082 11.4202188,8.46378132 L11.4202188,11.377344 C11.4202188,11.7779863 11.2805313,12.1208747 11.001,12.4062005 C10.7214688,12.6914624 10.3853438,12.8339818 9.99275,12.8339818 L2.5695,12.8339818 C2.1769375,12.8339818 1.84084375,12.6914624 1.56125,12.4062005 C1.28171875,12.1209066 1.14196875,11.7780182 1.14196875,11.377344 L1.14196875,3.80200456 C1.14196875,3.40139408 1.2816875,3.05853759 1.56125,2.77324374 C1.84084375,2.487918 2.1769375,2.34533485 2.5695,2.34533485 L8.8505625,2.34533485 C8.93396875,2.34533485 9.00240625,2.31794077 9.055875,2.26337585 C9.10934375,2.20874715 9.1360625,2.13900228 9.1360625,2.05394989 L9.1360625,1.47114806 C9.1360625,1.38625513 9.10934375,1.31638269 9.055875,1.26184966 C9.00240625,1.20722096 8.93396875,1.17995444 8.8505625,1.17995444 L2.5695,1.17995444 C1.86175,1.17995444 1.2565,1.43641913 0.7538125,1.94931663 C0.2513125,2.46218223 0,3.07984055 0,3.80210023 L0,11.3774715 C0,12.0996993 0.2513125,12.7174533 0.75384375,13.2301595 C1.25653125,13.7430888 1.86178125,13.9996811 2.56953125,13.9996811 L9.99275,13.9996811 C10.7004688,13.9996811 11.3057813,13.7430888 11.8084063,13.2301595 C12.3110625,12.7174533 12.562375,12.0997312 12.562375,11.3774715 L12.562375,8.46400456 C12.562375,8.37888838 12.535625,8.30904784 12.4819375,8.25448292 C12.4284063,8.19969476 12.3599688,8.17242825 12.2767813,8.17242825 Z"
                              id="Shape"
                            />
                            <path
                              d="M15.8009687,0.172974943 C15.6879375,0.0576264237 15.5539375,0 15.3994062,0 L10.8313437,0 C10.6767187,0 10.5428125,0.0576264237 10.429875,0.172974943 C10.3168438,0.288259681 10.26025,0.42484738 10.26025,0.58267426 C10.26025,0.740501139 10.3168438,0.877120729 10.429875,0.992469248 L12.0000938,2.59491116 L6.183,8.53138952 C6.1234375,8.59207745 6.09375,8.661918 6.09375,8.74068793 C6.09375,8.81974487 6.12353125,8.88955353 6.183,8.95024146 L7.200125,9.98809112 C7.25959375,10.048779 7.32803125,10.0790433 7.40534375,10.0790433 C7.48265625,10.0790433 7.551125,10.0488109 7.6105625,9.98809112 L13.4276875,4.05170843 L14.9980313,5.65415034 C15.1109687,5.76949886 15.2448438,5.82718907 15.3995,5.82718907 C15.5541562,5.82718907 15.6880312,5.76949886 15.8010937,5.65415034 C15.914125,5.53883371 15.9705937,5.4022779 15.9705937,5.24445103 L15.9705937,0.58267426 C15.9705625,0.424751708 15.9139063,0.288259681 15.8009687,0.172974943 Z"
                              id="Shape"
                            />
                          </g>
                        </g>
                      </g>
                    </svg>
                  </ExternalLink>
                </span>
              </a>
            </Link>
          </VirtualHearingLink>
          <div
            data-css-1abghdv=""
          >
            <strong>
              Conference Room: 
            </strong>
            BVA#00000001@localhost
          </div>
          <div
            data-css-6aejb2=""
          >
            <strong>
              PIN: 
            </strong>
            1234567890#
          </div>
          <CopyTextButton
            label=""
            text="Copy VLJ Link"
          >
            <Tooltip
              id="tooltip-Copy VLJ Link"
              position="bottom"
              text="Click to copy"
            >
              <span
                aria-describedby="tooltip-Copy VLJ Link"
                data-event="focus mouseenter"
                data-event-off="keydown mouseleave"
                data-for="tooltip-Copy VLJ Link"
                data-tip={true}
                role="tooltip"
                tabIndex={0}
              >
                <CopyToClipboard
                  text="Copy VLJ Link"
                >
                  <button
                    aria-label="Copy  Copy VLJ Link"
                    className="cf-apppeal-id"
                    data-css-cjlnsd=""
                    onClick={[Function]}
                    type="submit"
                  >
                    Copy VLJ Link
                     
                    <ClipboardIcon>
                      <svg
                        height="16"
                        viewBox="0 0 21 21"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M13.346 2.578h-2.664v-1.29C10.682.585 10.08 0 9.35 0H6.66c-.728 0-1.33.584-1.33 1.29v1.288H2.663v2.577h10.682V2.578zm-4.02 0H6.66v-1.29h2.665v1.29zm6.685 3.89V3.234a.665.665 0 0 0-.678-.656H14v1.29h.68v2.576H6.66v9.046H1.333V3.867h.68v-1.29H.678a.665.665 0 0 0-.68.657v12.913c0 .365.302.656.68.656h6.006v3.867h9.35l3.996-3.867V6.468h-4.02zm0 12.378v-2.043h2.112l-2.11 2.043zm2.665-3.356H14.68v3.867H7.992v-11.6h10.682v7.733z"
                          fill="#5B616B"
                          fillRule="evenodd"
                        />
                      </svg>
                    </ClipboardIcon>
                  </button>
                </CopyToClipboard>
              </span>
              <span
                data-css-15opktr=""
              >
                <ReactTooltip
                  clickable={false}
                  effect="solid"
                  id="tooltip-Copy VLJ Link"
                  insecure={true}
                  multiline={true}
                  offset={Object {}}
                  place="bottom"
                  resizeHide={true}
                  wrapper="div"
                >
                  <div
                    className="__react_component_tooltip place-bottom type-dark "
                    data-id="tooltip"
                    id="tooltip-Copy VLJ Link"
                  >
                    Click to copy
                  </div>
                </ReactTooltip>
              </span>
            </Tooltip>
          </CopyTextButton>
        </VirtualHearingLinkDetails>
      </div>
    </LinkContainer>
    <LinkContainer
      hearing={
        Object {
          "scheduledForIsPast": false,
        }
      }
      isVirtual={true}
      label="Guest Link"
      linkText="Join Virtual Hearing"
      role="Guest"
      user={
        Object {
          "addressCity": "Washington",
          "addressLine1": "123 Fake St.",
          "addressState": "DC",
          "addressZip": "20001",
          "name": "John Smith",
        }
      }
      virtualHearing={
        Object {
          "aliasWithHost": "BVA#00000001@localhost",
          "clientHost": "localhost",
          "guestPin": "12345678#",
          "hostPin": "1234567890#",
          "jobCompleted": true,
          "status": "active",
        }
      }
    >
      <div
        data-css-tx10g=""
        id="guest-hearings-link"
      >
        <strong>
          Guest Link
          : 
        </strong>
        <VirtualHearingLinkDetails
          aliasWithHost="BVA#00000001@localhost"
          hearing={
            Object {
              "scheduledForIsPast": false,
            }
          }
          isVirtual={true}
          label="Join Virtual Hearing"
          pin="12345678#"
          role="Guest"
          user={
            Object {
              "addressCity": "Washington",
              "addressLine1": "123 Fake St.",
              "addressState": "DC",
              "addressZip": "20001",
              "name": "John Smith",
            }
          }
          virtualHearing={
            Object {
              "aliasWithHost": "BVA#00000001@localhost",
              "clientHost": "localhost",
              "guestPin": "12345678#",
              "hostPin": "1234567890#",
              "jobCompleted": true,
              "status": "active",
            }
          }
        >
          <VirtualHearingLink
            hearing={
              Object {
                "scheduledForIsPast": false,
              }
            }
            isVirtual={true}
            label="Join Virtual Hearing"
            newWindow={true}
            showFullLink={false}
            user={
              Object {
                "addressCity": "Washington",
                "addressLine1": "123 Fake St.",
                "addressState": "DC",
                "addressZip": "20001",
                "name": "John Smith",
              }
            }
            virtualHearing={
              Object {
                "aliasWithHost": "BVA#00000001@localhost",
                "clientHost": "localhost",
                "guestPin": "12345678#",
                "hostPin": "1234567890#",
                "jobCompleted": true,
                "status": "active",
              }
            }
          >
            <Link
              disabled={false}
              target="_blank"
            >
              <a
                target="_blank"
                type={null}
              >
                <strong>
                  Join Virtual Hearing
                </strong>
                <span
                  data-css-32srrr=""
                >
                   
                  <ExternalLink
                    fill="#0071bc"
                  >
                    <svg
                      height="16px"
                      version="1.1"
                      viewBox="0 0 18 16"
                      width="18px"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g
                        fill="none"
                        fillRule="evenodd"
                        stroke="none"
                        strokeWidth="1"
                      >
                        <g
                          fill="#0071bc"
                          fillRule="nonzero"
                        >
                          <g
                            transform="translate(1.000000, 1.000000)"
                          >
                            <path
                              d="M12.2767813,8.17242825 L11.7057188,8.17242825 C11.6223125,8.17242825 11.553875,8.19969476 11.500625,8.25448292 C11.4469375,8.30895216 11.4202188,8.37876082 11.4202188,8.46378132 L11.4202188,11.377344 C11.4202188,11.7779863 11.2805313,12.1208747 11.001,12.4062005 C10.7214688,12.6914624 10.3853438,12.8339818 9.99275,12.8339818 L2.5695,12.8339818 C2.1769375,12.8339818 1.84084375,12.6914624 1.56125,12.4062005 C1.28171875,12.1209066 1.14196875,11.7780182 1.14196875,11.377344 L1.14196875,3.80200456 C1.14196875,3.40139408 1.2816875,3.05853759 1.56125,2.77324374 C1.84084375,2.487918 2.1769375,2.34533485 2.5695,2.34533485 L8.8505625,2.34533485 C8.93396875,2.34533485 9.00240625,2.31794077 9.055875,2.26337585 C9.10934375,2.20874715 9.1360625,2.13900228 9.1360625,2.05394989 L9.1360625,1.47114806 C9.1360625,1.38625513 9.10934375,1.31638269 9.055875,1.26184966 C9.00240625,1.20722096 8.93396875,1.17995444 8.8505625,1.17995444 L2.5695,1.17995444 C1.86175,1.17995444 1.2565,1.43641913 0.7538125,1.94931663 C0.2513125,2.46218223 0,3.07984055 0,3.80210023 L0,11.3774715 C0,12.0996993 0.2513125,12.7174533 0.75384375,13.2301595 C1.25653125,13.7430888 1.86178125,13.9996811 2.56953125,13.9996811 L9.99275,13.9996811 C10.7004688,13.9996811 11.3057813,13.7430888 11.8084063,13.2301595 C12.3110625,12.7174533 12.562375,12.0997312 12.562375,11.3774715 L12.562375,8.46400456 C12.562375,8.37888838 12.535625,8.30904784 12.4819375,8.25448292 C12.4284063,8.19969476 12.3599688,8.17242825 12.2767813,8.17242825 Z"
                              id="Shape"
                            />
                            <path
                              d="M15.8009687,0.172974943 C15.6879375,0.0576264237 15.5539375,0 15.3994062,0 L10.8313437,0 C10.6767187,0 10.5428125,0.0576264237 10.429875,0.172974943 C10.3168438,0.288259681 10.26025,0.42484738 10.26025,0.58267426 C10.26025,0.740501139 10.3168438,0.877120729 10.429875,0.992469248 L12.0000938,2.59491116 L6.183,8.53138952 C6.1234375,8.59207745 6.09375,8.661918 6.09375,8.74068793 C6.09375,8.81974487 6.12353125,8.88955353 6.183,8.95024146 L7.200125,9.98809112 C7.25959375,10.048779 7.32803125,10.0790433 7.40534375,10.0790433 C7.48265625,10.0790433 7.551125,10.0488109 7.6105625,9.98809112 L13.4276875,4.05170843 L14.9980313,5.65415034 C15.1109687,5.76949886 15.2448438,5.82718907 15.3995,5.82718907 C15.5541562,5.82718907 15.6880312,5.76949886 15.8010937,5.65415034 C15.914125,5.53883371 15.9705937,5.4022779 15.9705937,5.24445103 L15.9705937,0.58267426 C15.9705625,0.424751708 15.9139063,0.288259681 15.8009687,0.172974943 Z"
                              id="Shape"
                            />
                          </g>
                        </g>
                      </g>
                    </svg>
                  </ExternalLink>
                </span>
              </a>
            </Link>
          </VirtualHearingLink>
          <div
            data-css-1abghdv=""
          >
            <strong>
              Conference Room: 
            </strong>
            BVA#00000001@localhost
          </div>
          <div
            data-css-6aejb2=""
          >
            <strong>
              PIN: 
            </strong>
            12345678#
          </div>
          <CopyTextButton
            label=""
            text="Copy Guest Link"
          >
            <Tooltip
              id="tooltip-Copy Guest Link"
              position="bottom"
              text="Click to copy"
            >
              <span
                aria-describedby="tooltip-Copy Guest Link"
                data-event="focus mouseenter"
                data-event-off="keydown mouseleave"
                data-for="tooltip-Copy Guest Link"
                data-tip={true}
                role="tooltip"
                tabIndex={0}
              >
                <CopyToClipboard
                  text="Copy Guest Link"
                >
                  <button
                    aria-label="Copy  Copy Guest Link"
                    className="cf-apppeal-id"
                    data-css-cjlnsd=""
                    onClick={[Function]}
                    type="submit"
                  >
                    Copy Guest Link
                     
                    <ClipboardIcon>
                      <svg
                        height="16"
                        viewBox="0 0 21 21"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M13.346 2.578h-2.664v-1.29C10.682.585 10.08 0 9.35 0H6.66c-.728 0-1.33.584-1.33 1.29v1.288H2.663v2.577h10.682V2.578zm-4.02 0H6.66v-1.29h2.665v1.29zm6.685 3.89V3.234a.665.665 0 0 0-.678-.656H14v1.29h.68v2.576H6.66v9.046H1.333V3.867h.68v-1.29H.678a.665.665 0 0 0-.68.657v12.913c0 .365.302.656.68.656h6.006v3.867h9.35l3.996-3.867V6.468h-4.02zm0 12.378v-2.043h2.112l-2.11 2.043zm2.665-3.356H14.68v3.867H7.992v-11.6h10.682v7.733z"
                          fill="#5B616B"
                          fillRule="evenodd"
                        />
                      </svg>
                    </ClipboardIcon>
                  </button>
                </CopyToClipboard>
              </span>
              <span
                data-css-4jj7wk=""
              >
                <ReactTooltip
                  clickable={false}
                  effect="solid"
                  id="tooltip-Copy Guest Link"
                  insecure={true}
                  multiline={true}
                  offset={Object {}}
                  place="bottom"
                  resizeHide={true}
                  wrapper="div"
                >
                  <div
                    className="__react_component_tooltip place-bottom type-dark "
                    data-id="tooltip"
                    id="tooltip-Copy Guest Link"
                  >
                    Click to copy
                  </div>
                </ReactTooltip>
              </span>
            </Tooltip>
          </CopyTextButton>
        </VirtualHearingLinkDetails>
      </div>
    </LinkContainer>
    <div />
  </div>
</HearingLinks>
`;

exports[`HearingLinks Matches snapshot when hearing was virtual and occurred 1`] = `
<HearingLinks
  hearing={
    Object {
      "scheduledForIsPast": false,
    }
  }
  user={
    Object {
      "addressCity": "Washington",
      "addressLine1": "123 Fake St.",
      "addressState": "DC",
      "addressZip": "20001",
      "name": "John Smith",
    }
  }
  virtualHearing={
    Object {
      "aliasWithHost": "BVA#00000001@localhost",
      "clientHost": "localhost",
      "guestPin": "12345678#",
      "hostPin": "1234567890#",
      "jobCompleted": true,
      "status": "active",
    }
  }
  wasVirtual={true}
>
  <div
    data-css-19plkw7=""
    data-css-1fmltfu=""
  >
    <LinkContainer
      hearing={
        Object {
          "scheduledForIsPast": false,
        }
      }
      label="VLJ Link"
      linkText="Start Virtual Hearing"
      role="VLJ"
      user={
        Object {
          "addressCity": "Washington",
          "addressLine1": "123 Fake St.",
          "addressState": "DC",
          "addressZip": "20001",
          "name": "John Smith",
        }
      }
      virtualHearing={
        Object {
          "aliasWithHost": "BVA#00000001@localhost",
          "clientHost": "localhost",
          "guestPin": "12345678#",
          "hostPin": "1234567890#",
          "jobCompleted": true,
          "status": "active",
        }
      }
      wasVirtual={true}
    >
      <div
        data-css-tx10g=""
        id="vlj-hearings-link"
      >
        <strong>
          VLJ Link
          : 
        </strong>
        <VirtualHearingLinkDetails
          aliasWithHost="BVA#00000001@localhost"
          hearing={
            Object {
              "scheduledForIsPast": false,
            }
          }
          label="Start Virtual Hearing"
          pin="1234567890#"
          role="VLJ"
          user={
            Object {
              "addressCity": "Washington",
              "addressLine1": "123 Fake St.",
              "addressState": "DC",
              "addressZip": "20001",
              "name": "John Smith",
            }
          }
          virtualHearing={
            Object {
              "aliasWithHost": "BVA#00000001@localhost",
              "clientHost": "localhost",
              "guestPin": "12345678#",
              "hostPin": "1234567890#",
              "jobCompleted": true,
              "status": "active",
            }
          }
          wasVirtual={true}
        >
          <span>
            Expired
          </span>
          <div
            data-css-1abghdv=""
          >
            <strong>
              Conference Room: 
            </strong>
            BVA#00000001@localhost
          </div>
          <div
            data-css-6aejb2=""
          >
            <strong>
              PIN: 
            </strong>
            1234567890#
          </div>
        </VirtualHearingLinkDetails>
      </div>
    </LinkContainer>
    <LinkContainer
      hearing={
        Object {
          "scheduledForIsPast": false,
        }
      }
      label="Guest Link"
      linkText="Join Virtual Hearing"
      role="Guest"
      user={
        Object {
          "addressCity": "Washington",
          "addressLine1": "123 Fake St.",
          "addressState": "DC",
          "addressZip": "20001",
          "name": "John Smith",
        }
      }
      virtualHearing={
        Object {
          "aliasWithHost": "BVA#00000001@localhost",
          "clientHost": "localhost",
          "guestPin": "12345678#",
          "hostPin": "1234567890#",
          "jobCompleted": true,
          "status": "active",
        }
      }
      wasVirtual={true}
    >
      <div
        data-css-tx10g=""
        id="guest-hearings-link"
      >
        <strong>
          Guest Link
          : 
        </strong>
        <VirtualHearingLinkDetails
          aliasWithHost="BVA#00000001@localhost"
          hearing={
            Object {
              "scheduledForIsPast": false,
            }
          }
          label="Join Virtual Hearing"
          pin="12345678#"
          role="Guest"
          user={
            Object {
              "addressCity": "Washington",
              "addressLine1": "123 Fake St.",
              "addressState": "DC",
              "addressZip": "20001",
              "name": "John Smith",
            }
          }
          virtualHearing={
            Object {
              "aliasWithHost": "BVA#00000001@localhost",
              "clientHost": "localhost",
              "guestPin": "12345678#",
              "hostPin": "1234567890#",
              "jobCompleted": true,
              "status": "active",
            }
          }
          wasVirtual={true}
        >
          <span>
            Expired
          </span>
          <div
            data-css-1abghdv=""
          >
            <strong>
              Conference Room: 
            </strong>
            BVA#00000001@localhost
          </div>
          <div
            data-css-6aejb2=""
          >
            <strong>
              PIN: 
            </strong>
            12345678#
          </div>
        </VirtualHearingLinkDetails>
      </div>
    </LinkContainer>
    <div />
  </div>
</HearingLinks>
`;

exports[`HearingLinks Matches snapshot with default props when passed in 1`] = `<HearingLinks />`;
